src/main/java/com/google/devtools/build/lib/packages/StructImpl.java

```diff
@@ -333,10 +327,16 @@ private void printJson(Object value, StringBuilder sb, Location loc, String cont
     }
   }
 
-  private String jsonEscapeString(String string) {
-    return escapeDoubleQuotesAndBackslashesAndNewlines(string)
+  private void jsonEscapeString(StringBuilder stringBuilder, String string, boolean appendColon) {
```

Comment:
Let's call this function appendJSONStringLiteral, the StringBuilder 'out', and the String s.

Reply:
Sounds good.